src/Proto.Remote/RemoteProcess.cs

```diff
@@ -41,7 +39,7 @@ private void Send(object msg)
 
         private void SendMessage(object msg)
         {
-            _remote.SendMessage(_pid, msg, -1);
+            _endpointManager.SendMessage(_pid, msg, -1);

```

Comment:
Looking at this, makes me wonder if we somehow safely can bypass the `EnsureConnected` part in endpoint manager SendMessage/RemoteDeliver.
if that check is done once and then return some stateful object back to the RemoteProcess, which then holds on to that.
We could likely improve remote throughput by avoiding the dictionary lookup for address for each message.

Not a critical fix for this PR, but still worth thinking about.